ietf/meeting/views.py

```diff
@@ -2568,6 +2569,66 @@ def add_session_drafts(request, session_id, num):
                     'form': form,
                   })
 
+class SessionNotesAndRecordingsForm(forms.Form):
+    title = forms.CharField(max_length=255)
+    url = forms.URLField(label="Link to recording (YouTube only)")
+
+    def __init__(self, *args, **kwargs):
+        self.already_linked = kwargs.pop('already_linked')
+        super(self.__class__, self).__init__(*args, **kwargs)
+
+    def __init__(self, *args, **kwargs):
+        self.already_linked = kwargs.pop('already_linked')
+        super(self.__class__, self).__init__(*args, **kwargs)
+
+    def clean_url(self):
+        url = self.cleaned_data['url']
+        parsed_url = urlparse(url)
+        if parsed_url.hostname not in YOUTUBE_DOMAINS:
+            raise forms.ValidationError("Must be a YouTube URL")
+        problems = set(url).intersection(set(self.already_linked)) 
+        if problems:
+           raise forms.ValidationError("Already linked: %s" % ', '.join([d.name for d in problems]))
+        return url
+
+
+def add_session_recordings(request, session_id, num):
+    # num is redundant, but we're dragging it along an artifact of where we are in the current URL structure
+    session = get_object_or_404(Session,pk=session_id)
+    if not session.can_manage_materials(request.user):
+        raise Http404
+    if session.is_material_submission_cutoff() and not has_role(request.user, "Secretariat"):
+        raise Http404
+
+    already_linked = [material.external_url for material in session.materials.filter(type="recording").exclude(states__type="recording", states__slug='deleted').order_by('presentations__order')]
+
+    session_number = None
+    sessions = get_sessions(session.meeting.number,session.group.acronym)
+    if len(sessions) > 1:
+       session_number = 1 + sessions.index(session)
+
+    if request.method == 'POST':
+        delete = request.POST['delete']
+        if delete:
+            delete_recording(pk=delete, session=session)
```

Comment:
>  We should think through what a chair would do if they entered the wrong link and wanted to fix it. @rjsparks 

I've added a delete button. Is that ok?


![Screenshot from 2024-11-18 10-56-26](https://github.com/user-attachments/assets/82295bc2-73b2-46c6-9a0b-864db4d140e4)

Reply:
Regarding the 'Revision' column is that valid for recordings? How would someone update the revision of a recording anyway?

Reply:
It's valid as long as we're modeling recordings as Document objects.

To date, we have not revised a recording document. Doesn't mean we couldn't (with the current model anyhow). An -01 for youtube would just point to a different URL the same as an -01 for a draft points to a different filename.

Reply:
Thoughts on the delete button? Should it have a confirmation modal?

Reply:
I'm going to assume we need a confirmation modal.

Reply:
I have minor heartburn that we're allowing a Document object to be deleted. It violates an assumption about what Document objects are _for_, but I suppose we have that problem with these recording documents-that-are-just-external-links already and some future project should migrate them to some other model instead.
 
I'll request one change and we can move forward with this.